plugins/techdocs/src/reader/components/Reader.tsx

```diff
@@ -155,7 +155,7 @@ export const Reader = ({ entityId, onReady }: Props) => {
   ]);
 
   if (error) {
-    return <TechDocsNotFound errorMessage={error.message} />;
+    return <TechDocsNotFound statusCode={404} errorMessage={error.message} />;
```

Comment:
At this point, a shadowDom has been calculcated after being received back from the API call.  I could not figure out how to get the statusCode from the resulting API call, passed through to this method here, and therefore to change this line.  Ideally, it should have been something like:

```typescript
return <TechDocsNotFound statusCode={returnedApiStatusCode} errorMessage{errror.message} />;
```

But since I couldn't collect it, I just hardcoded it as the 404 into this component.  If you had any suggestions for this Reader method I'd welcome it!

Reply:
This may be where my lack of TypeScript expertise as a non-programmer rears it's ugly face. I guess some way to collect from this call?

https://github.com/backstage/backstage/blob/45de779d5fa75721dd8bb1865c3e63a8ba8c4976/plugins/techdocs/src/reader/components/Reader.tsx#L53-L55

Reply:
Yeah! IMO we should be throwing separate custom error types from the API logic
https://github.com/backstage/backstage/blob/45de779d5fa75721dd8bb1865c3e63a8ba8c4976/plugins/techdocs/src/api.ts#L124-L135

and use the error type to display the right error message. I am not sure if the `@backstage` frontend package provides custom error types like the backend has 

https://github.com/backstage/backstage/blob/435972704c9acf85fda79e28c9556482780c801d/packages/backend-common/src/errors.ts#L50-L82

@Rugvip @freben I'll be thankful for your comments on this!

Reply:
Wonder if this might be ok for an incremental improvement without the custom error throws, and then increment later. Key for me is avoiding just the raw msg being printed like so:

![image](https://user-images.githubusercontent.com/33203301/106994451-c8db7b00-674a-11eb-87b4-ffb908204d6b.png)

And this PR does accomplish that?  I could add a comment into where I hardcode the 404 and say why with a TODO? What do you think @OrkoHunter ?

Reply:
A TODO about 404 sounds fine @adamdmharvey. :)